### PR TITLE
Fix/issue#32

### PR DESCRIPTION
**Pull request by erisara**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -17,7 +17,16 @@
...
-
+		<dependency>
+			<groupId>net.sf.ehcache</groupId>
+			<artifactId>ehcache</artifactId>
+			<version>2.8.3</version>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-context-support</artifactId>
+			<version>4.1.0.RELEASE</version>
```
Comment (dionisisvas) [inline pom.xml]: Use ${spring.version} to keep consistency with the rest of the spring dependencies

File: src/main/java/com/iri/training/repository/ConnectToBase.java [hunk 2]
```diff
@@ -1,31 +1,22 @@
...
-	private static Connection c;
-
-	public Connection getConnection() throws SQLException {
-
-		try {
-			Class.forName("org.sqlite.JDBC");
-			c = DriverManager.getConnection("jdbc:sqlite:db\\TrainingApp.db");
-			System.out.println("Opened database successfully");
 
+		@Bean
```
Comment (dionisisvas) [inline src/main/java/com/iri/training/repository/ConnectToBase.java]: Try to put '@Bean' annotation before class declaration and autowire it to to the repositories needed

File: src/main/java/com/iri/training/repository/ConnectToBase.java [hunk 3]
```diff
@@ -1,31 +1,22 @@
...
+		public static DriverManagerDataSource getDataSource() {
```
Comment (dionisisvas) [inline src/main/java/com/iri/training/repository/ConnectToBase.java]: You can remove static declaration if the comment above is implemented

File: src/main/java/com/iri/training/web/service/UserServiceImpl.java [hunk 4]
```diff
@@ -16,11 +16,11 @@
 	
 	@Override
 	public User getUserById(Long userId) throws SQLException {
-		
+		//TODO: logs
```
Comment (dionisisvas) [inline src/main/java/com/iri/training/web/service/UserServiceImpl.java]: Add the missing logs

File: pom.xml [hunk 5]
```diff
@@ -35,7 +55,11 @@
     		<artifactId>spring-jdbc</artifactId>
     		<version>${spring.version}</version>
 		</dependency>
-
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-autoconfigure</artifactId>
+			<version>1.5.3.RELEASE</version>
```
Comment (dionisisvas) [inline pom.xml]: Use ${spring.version} as in spring dependencies above

File: pom.xml [hunk 6]
```diff
@@ -16,8 +16,28 @@
...
+	<dependencies>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-core</artifactId>
+			<version>2.4.3</version>
+		</dependency>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-databind</artifactId>
+			<version>2.4.3</version>
```
Comment (dionisisvas) [inline pom.xml]: Where is jackson-? dependency used? is it really needed?

**Review by dionisisvas — COMMENTED** [review]
Check comments

File: src/main/java/com/iri/training/repository/ConnectToBase.java (hunk 3 above)
Comment (dionisisvas) [reply src/main/java/com/iri/training/repository/ConnectToBase.java]: I am not sure about the above, try it and if it does not work you can leave as it is

File: pom.xml (hunk 6 above)
Comment (erisara) [inline pom.xml]: Without this dependency it shows this message: No converter found for return value of type: class com.iri.training.model.User

File: src/main/java/com/iri/training/web/service/UserServiceImpl.java [hunk 7]
```diff
@@ -2,32 +2,36 @@
...
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import com.iri.training.model.User;
+import com.iri.training.repository.UserCommentRepositoryImpl;
 import com.iri.training.repository.UserRepository;
 
 @Service
 public  class UserServiceImpl implements UserService {
+	Logger logger = Logger.getLogger(UserCommentRepositoryImpl.class);
```
Comment (dionisisvas) [inline src/main/java/com/iri/training/web/service/UserServiceImpl.java]: Use the correct class in logger: Logger logger = Logger.getLogger(UserServiceImpl.class);

**Pull request merged**